lighthouse-core/report/v2/report-renderer.js

```diff
@@ -81,39 +96,104 @@ window.ReportRenderer = class ReportRenderer {
+    });
     return element;
   }
 
   /**
+   * @param {string} selector
+   * @return {!DocumentFragment} A clone of the template content.
+   * @throws {Error}
+   */
+  cloneTemplate(selector) {
+    const template = this._document.querySelector(selector);
+    if (!template) {
+      throw new Error(`Template not found: template${selector}`);
+    }
+    return this._document.importNode(template.content, true);
+  }
+}
+
+class ReportRenderer {
+  /**
+   * @param {!Document} document
+   */
+  constructor(document) {
+    this._dom = new DOM(document);
+  }
+
+  /**
+   * @param {!ReportJSON} report
+   * @return {!Element}
+   */
+  renderReport(report) {
+    try {
+      return this._renderReport(report);
+    } catch (e) {
+      return this._renderException(e);
+    }
+  }
+
+  /**
+   * @param {!DocumentFragment|Element} element DOM node to populate with values.
    * @param {number} score
+   * @param {string} scoringMode
    * @param {string} title
    * @param {string} description
    * @return {!Element}
    */
-  _renderScore(score, title, description) {
-    const element = this._createElement('div', 'lighthouse-score');
-
-    const value = element.appendChild(this._createElement('div', 'lighthouse-score__value'));
-    value.textContent = formatNumber(score);
-    value.classList.add(`lighthouse-score__value--${calculateRating(score)}`);
+  _populateResult(element, score, scoringMode, title, description) {
+    // Fill in the blanks.
+    const valueEl = element.querySelector('.lighthouse-score__value');
+    DOM.setText(valueEl, formatNumber(score));
+    DOM.addClass(valueEl, `lighthouse-score__value--${calculateRating(score)}`,
+        `lighthouse-score__value--${scoringMode}`);
 
-    const column = element.appendChild(this._createElement('div', 'lighthouse-score__text'));
-    column.appendChild(this._createElement('div', 'lighthouse-score__title')).textContent = title;
-    column.appendChild(
-      this._createElement('div', 'lighthouse-score__description')).textContent = description;
+    DOM.setText(element.querySelector('.lighthouse-score__title'), title);
+    DOM.setText(element.querySelector('.lighthouse-score__description'), description);
 
     return element;
   }
 
   /**
+   * @param {!AuditJSON} audit
+   * @return {!Element}
+   */
+  _renderAuditScore(audit) {
+    const tmpl = this._dom.cloneTemplate('#tmpl-lighthouse-audit-score');
+
+    const scoringMode = audit.result.scoringMode;
+    const description = audit.result.helpText;
+    let title = audit.result.description;
+
+    if (audit.result.displayValue) {
+      title += `:  ${audit.result.displayValue}`;
+    }
+    if (audit.result.optimalValue) {
+      title += ` (target: ${audit.result.optimalValue})`;
+    }
+
+    return this._populateResult(tmpl, audit.score, scoringMode, title, description);
+  }
+
+  /**
+   * @param {!CategoryJSON} category
+   * @return {!Element}
+   */
+  _renderCategoryScore(category) {
```

Comment:
> I'm actually going to change `_populateResult` back to `_populateScore`. It's populating the score dom with values

my issue was that if someone was going into the codebase to see where audits' titles and descriptions are set, they probably wouldn't think it would be in `_populateScore`. If the issue is the template's id, that can change too :P

This is all going to change a lot in the near future, so if you feel strongly we can defer